### PR TITLE
Rework Flock conversion code

### PR DESCRIPTION
**Pull request by FlameArrow57**
[INTERNAL][REWORK]

## About the PR 
This PR reworks how Flock conversion is coded.

## Why's this needed? 
Makes it more maintainable

### CODE REVIEW

File: code/datums/flock/flock.dm [hunk 1]
```diff
@@ -703,6 +686,51 @@ var/flock_signal_unleashed = FALSE
...
+		for (var/subtype in typesof(/obj/storage/secure/closet))
+			flock_conversion_paths[subtype] = /obj/storage/closet/flock
+		for (var/subtype in typesof(/obj/machinery/computer))
+			flock_conversion_paths[subtype] = /obj/flock_structure/compute
+		for (var/subtype in typesof(/obj/machinery/computer3))
+			flock_conversion_paths[subtype] = /obj/flock_structure/compute
+		for (var/subtype in typesof(/obj/machinery/networked/teleconsole))
+			flock_conversion_paths[subtype] = /obj/flock_structure/compute
+		for (var/subtype in typesof(/obj/machinery/networked/mainframe))
+			flock_conversion_paths[subtype] = /obj/flock_structure/compute/mainframe
```
Comment (TobleroneSwordfish) [inline code/datums/flock/flock.dm:717]: This just seems less maintainable than what was already there?

File: code/datums/flock/flock.dm [hunk 2]
```diff
@@ -703,6 +686,51 @@ var/flock_signal_unleashed = FALSE
...
+
+		var/types_to_skip = list(/obj/machinery/computer/card/portable, /obj/machinery/computer/security/wooden_tv,
```
Comment (TobleroneSwordfish) [inline code/datums/flock/flock.dm:719]: this is better, but I still think it would make sense as a check than maintaining a list of hundreds of subtypes

**Review by TobleroneSwordfish — CHANGES_REQUESTED** [review]
I get what you're trying to do, but I feel like making a list of every subtype of door, window and table is a bad idea for performance and maintainability reasons

**Comment by FlameArrow57** [comment]
Yeah think I overlooked that aspect, will try to revise if I can

**Pull request closed without merging**